docroot/modules/humsci/hs_person/src/Service/PersonAuthorship.php

```diff
@@ -0,0 +1,466 @@
+   *
+   * @param array $nodes
+   *   Array of node entities that were assigned.
+   * @param string|null $user_email
+   *   The email address used for matching.
+   * @param string|null $authname
+   *   The user's authname (SUNet ID) used for matching.
+   */
+  protected function notifyUser(array $nodes, ?string $user_email, ?string $authname = NULL): void {
+    $matching_nodes = array_filter($nodes, function ($node) use ($user_email, $authname) {
+      return $this->isNodeMatch($node, $user_email, $authname);
+    });
+
+    if (empty($matching_nodes)) {
+      return;
+    }
+
+    $node_count = count($matching_nodes);
+    $node_links = array_map(function ($node) {
+      return Markup::create('<a href="' . $node->toUrl()->toString() . '">' . $node->getTitle() . '</a>');
+    }, $matching_nodes);
+
+    if ($node_count === 1) {
+      $this->messenger->addStatus(
+        t('A matching person profile was found for @node_link. You now have permission to edit it.', [
+          '@node_link' => reset($node_links),
+        ])
+      );
+    }
+    else {
+      $this->messenger->addStatus(
+        t('@count matching person profiles were found. You now have permission to edit: @node_links', [
+          '@count' => $node_count,
+          '@node_links' => Markup::create(implode(', ', $node_links)),
+        ])
+      );
+    }
+  }
+
+  /**
+   * Get the email address for a user account.
+   *
+   * For SSO users, look up their authmap.authname and appends the SSO domain.
+   * For regular users, this returns their email address as a fallback.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The user account.
+   *
+   * @return string|null
+   *   The email address or null if none found.
+   */
+  protected function getUserEmail(AccountInterface $account): ?string {
+    // First, check if this is an SSO user by getting their authname.
+    $authname = $this->getUserAuthname($account);
+    if ($authname) {
+      $sso_email = $authname . self::SSO_DOMAIN;
+      return $sso_email;
+    }
+
+    // If no SSO authname, try to get the user's email address directly.
+    $email = $account->getEmail();
+    if (!empty($email)) {
+      return $email;
+    }
+
+    // No email found.
+    return NULL;
+  }
+
+  /**
+   * Get the authname (SUNet ID) for a user account.
+   *
+   * @todo Update this method to use a service from stanford_samlauth when one
+   *   becomes available, instead of directly querying the authmap table.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The user account.
+   *
+   * @return string|null
+   *   The authname or null if none found.
+   */
+  protected function getUserAuthname(AccountInterface $account) {
+    // First, check if this is an SSO user by looking up authmap.
+    $authname = $this->database->select('authmap', 'a')
+      ->fields('a', ['authname'])
+      ->condition('uid', $account->id())
+      ->execute()
+      ->fetchField();
```

Comment:
There is actually a service we should use for this:

- In the externalauth module there is an authmap service: `externalauth.authmap`
- This service has a get method: `public function get(int $uid, string $provider) {`
- This will get the authname based on the `uid` and `provider`.
- The provider in this case is "samlauth"
- When injected should be an easy call:
`$authname = $this->authmap->get($account->id(), 'samlauth');`

Reply:
The `hs_person.info.yml` should also probably have a dependency on `stanford_samlauth` or `samlauth` itself.